### ISSUE TITLE
Running on GPU

### ISSUE THREAD

**Issue by zhj2022** [huawei-noah/trustworthyAI#140]
The paper "gCastle: A Python Toolbox for Causal Discovery" claims that "gCastle includes ... with **optional GPU acceleration**". However, I don't know how GPU acceleration can be used on this package. Can you give me an example of its usage?

**Comment by shaido987** [comment]
You are correct. The input will directly be used for setting the `CUDA_VISIBLE_DEVICES` environment variable so setting "0, 1" should work. 

Following https://discuss.pytorch.org/t/os-environ-cuda-visible-devices-not-functioning/105545, you can try to set it yourself before doing any other imports at the top of the script (and just keep `device_ids` as the default). It could be that the setting is not taking effect correctly.

**Comment by zhj2022** [comment]
Yes, setting the environment variable `CUDA_VISIBLE_DEVICES` on the top can help me determine which gpu to use. For example, if I have a machine on which two gpus are available,  the command `os.environ["CUDA_VISIBLE_DEVICES"] = "1"` or `os.environ["CUDA_VISIBLE_DEVICES"] = "1, 0"` can make the program run on `cuda:1`.

However, the source code of the algorithm have the following commands:
```python
if self.device_type == 'gpu':
    if self.device_ids:
        os.environ['CUDA_VISIBLE_DEVICES'] = str(self.device_ids)
    device = torch.device('cuda')
else:
    device = torch.device('cpu')
```

In this way, it seems that the only thing I can do is decide which gpu to use, while training a single model on multiple gpus is not feasible.

**Comment by shaido987** [comment]
Can you help confirm if running the following works for multiple gpus? I.e., setting `CUDA_VISIBLE_DEVICES` before the import as in the link above?

```python
os.environ["CUDA_VISIBLE_DEVICES"] = "0, 1"

from castle.algorithms import RL
rl = RL(device_type='gpu', device_ids=None)
```

The above should run on both gpu 0 and 1.
